src/Async/README.md

```diff
@@ -0,0 +1,1369 @@
+```
+
+#### alt
+
+```haskell
+Async e a ~> Async e a -> Async e a
+```
+
+Providing a means for a fallback or alternative value, `alt` combines (2)
+`Async` instances and will return the first `Resolved` instance it encounters
+or the last `Rejected` instance if it does not encounter a `Resolved` instance.
+
+<!-- eslint-disable no-console -->
+<!-- eslint-disable no-sequences -->
+```javascript
+const { Rejected, Resolved } = require('crocks/Async')
+
+// log :: String -> a -> a
+const log = label => x =>
+  (console.log(`${label}:`, x), x)
+
+Resolved(true)
+  .alt(Rejected('Bad News'))
+  .fork(log('rej'), log('res'))
+//=> res: true
+
+Rejected('First Reject')
+  .alt(Rejected('Second Reject'))
+  .fork(log('rej'), log('res'))
+//=> rej: "Second Reject"
+
+Rejected('First Reject')
+  .alt(Resolved('First Resolve'))
+  .alt(Rejected('Second Reject'))
+  .alt(Resolved('Second Resolve'))
+  .fork(log('rej'), log('res'))
+//=> rej: "First Resolve"
+```
+
+#### bimap
+
+```haskell
+Async e a ~> ((e -> b), (a -> c)) -> Async b c
+```
+
+Both `Rejected` and `Resolved` values can vary in their type, although most of
+the time, focus on mapping values is placed on the `Resolved` portion. When the
+requirement or need to map the `Rejected` portion arises, `bimap` can be
+used.
+
+`bimap` takes (2) functions as its arguments. The first function is used
+to map a `Rejected` instance, while the second maps a `Resolved` instance.
+While `bimap` requires that both possible instances are to be mapped, if the
+desire to map only the `Rejected` portion, an `identity` function can be
+provided to the second argument. This will leave all `Resolved` instance values
+untouched.
+
+<!-- eslint-disable no-console -->
+<!-- eslint-disable no-sequences -->
+```javascript
+const { Rejected, Resolved } = require('crocks/Async')
+
+const assoc = require('crocks/helpers/assoc')
+const bimap = require('crocks/pointfree/bimap')
+const compose = require('crocks/helpers/compose')
+const objOf = require('crocks/helpers/objOf')
+
+// log :: String -> a -> a
+const log = label => x =>
+  (console.log(`${label}:`, x), x)
+
+// hasError :: Boolean -> Object -> Object
+const hasError =
+  assoc('hasError')
+
+// buildError :: a -> String
+const buildError =
+  x => `${x}: is not valid`
+
+// buildResult :: (String, Boolean) -> a -> Object
+const buildResult = (key, isError) =>
+  compose(hasError(isError), objOf(key))
+
+// finalize :: Bifunctor m => m a b -> m Object Object
+const finalize = bimap(
+  compose(buildResult('error', true), buildError),
+  buildResult('result', false)
+)
+
+finalize(Resolved('Good To Go'))
+  .fork(log('rej'), log('res'))
+//=> res: { result: "Good To Go", hasError: false }
+
+finalize(Rejected(null))
+  .fork(log('rej'), log('res'))
+//=> rej: { error: "null is not valid", hasError: true }
+```
+
+#### ap
+
+```haskell
+Async e (a -> b) ~> Async e a -> Async e b
+```
+
+Short for apply, `ap` is used to apply an `Async` instance containing a value
```

Comment:
Any reason to not use `apply` instead of `ap`? IMHO `ap` can be confused. (Probably this is not related to that PR)

Reply:
`ap` is defined by Fantasy-Land. Say we want to have "static" functions on our constructor (we do not do this in crocks, but some libs do), `apply` would override the `Function.prototype.apply`. Even though we do not implement the `ap` spec as defined by Fantasy-Land, it makes sense to still keep the naming 🌽sistant

Reply:
Yeah, was thinking something about the `apply` method, but since was an object I asked it. But yeah, now make sense. 👍